modules/minimint-ln/src/lib.rs

```diff
@@ -398,13 +375,46 @@ impl FederationModule for LightningModule {
             })
             .into_group_map();
 
-        for (contract_id, shares) in preimage_decraption_shares {
+        let mut bad_peers = vec![];
+        for (contract_id, shares) in preimage_decryption_shares {
             let span = info_span!("decrypt_preimage", %contract_id);
             let _gaurd = span.enter();
 
-            if shares.len() < self.cfg.threshold {
-                trace!(
-                    shares = %shares.len(),
+            let contract: ContractAccount = match self.get_contract_account(contract_id) {
+                Some(contract) => contract,
+                None => {
+                    warn!("Received decryption share for non-existent contract");
+                    continue;
+                }
+            };
+
+            let incoming_contract = match contract.contract {
+                FundedContract::Incoming(incoming) => incoming.contract,
+                _ => {
+                    warn!("Received decryption share for unfunded contract");
+                    continue;
```

Comment:
If we first always save the shares (irrespective of validity) and then just bail here if the contract doesn't exist we will never clean the shares from our DB (see #181). We need to either delete shares when we notice they are invalid or not save them in the first place (if the contract doesn't exist that might be ok, we don't mark peers proposing these shares as faulty anyway).

Reply:
I think this should never happen under normal circumstances, since peers shouldn't propose decryption shares until the contract is funded in `apply_output`, so I can delete the shares if this happens.  Maybe @justinmoon can confirm whether this is okay to do.

Reply:
The problem is we don't check decryption shares we receive from consensus before writing to DB: https://github.com/fedimint/minimint/pull/186/files#diff-3877e745c3cf511913784efe63b090c6a95046d0492b5999224631a2a0d0eb83R163-R167

That means if a malicious peer contributes a share not belonging to a contract it will never be cleaned up iiuc.

Reply:
Sorry I should have been more clear, I'm cleaning those up in `end_consensus_epoch` with the latest code I pushed now.

As I can tell this wouldn't clean up useful shares, but was wondering if anyone thought otherwise.

Reply:
Ah, sorry, didn't notice!